backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/AzureBackupStore.java

```diff
@@ -24,16 +32,62 @@
  * scheme: {@code basePath/partitionId/checkpointId/nodeId}.
  */
 public final class AzureBackupStore implements BackupStore {
-
-  private final AzureBackupConfig azureBackupConfig;
+  public static final String SNAPSHOT_FILESET_NAME = "snapshot";
+  public static final String SEGMENTS_FILESET_NAME = "segments";
+  private final ExecutorService executor;
+  private final FileSetManager fileSetManager;
+  private final ManifestManager manifestManager;
+  private final BlobContainerClient blobContainerClientManifests;
+  private final BlobContainerClient blobContainerClientContents;
 
   public AzureBackupStore(final AzureBackupConfig config) {
-    azureBackupConfig = config;
+    this(config, buildClient(config));
+  }
+
+  public AzureBackupStore(final AzureBackupConfig config, final BlobServiceClient client) {
+    executor = Executors.newVirtualThreadPerTaskExecutor();
+    blobContainerClientManifests =
+        client.getBlobContainerClient(config.containerName() + "-manifests");
+    blobContainerClientContents =
+        client.getBlobContainerClient(config.containerName() + "-contents");
```

Comment:
:wrench: Same here, use `/` as the separator, not `-`

Reply:
For the container name, some characters are forbidden such as "_" or using the "/"

Reply:
Oh interesting! So far I thought that containers are just an abstraction around normal object keys, but that seems to be not the case since object keys may contain `/`.

Reply:
I'd still prefer we only have one container, based on `config.containerName`.
Since containers are an actual thing for Azure Blob Storage, it'd be easier if we just use one container with a predicable naming. This way, the container can be set up ahead of time and our client doesn't need permissions to create arbitrary containers at runtime. Additionally, we stay closer to the other stores and don't have to worry about "ugly" container names such as "-contents".

I think this should be a relatively easy change, let me know if it isn't @rodrigo-lourenco-lopes